vividus-plugin-rest-api/src/main/java/org/vividus/http/CurlUtils.java

```diff
@@ -0,0 +1,108 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.http;
+
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpRequest;
+
+public final class CurlUtils
+{
+    private static final String CURL_REQUEST_COMMAND = "curl -X %s '%s' %n";
+    private static final String HEADER = "-H '%s: %s' %n";
+    private static final String BODY = "-d '%s'";
+    private static final String FORM_TEXT = "-F \"%s=%s\" %n";
+    private static final String FORM_FILE = "-F \"%s=@<path-to-file>%s\" %n";
+
+    private static final Pattern NAME_FILE_NAME_PATTERN = Pattern.compile("name=\"(.+)\"; filename=\"(.+)\"");
+    private static final Pattern NAME_CONTENT_PATTERN = Pattern.compile("name=\"(.+)\"\nContent-Type:.+\n\n(.*)");
+
+    private CurlUtils()
+    {
+    }
+
+    public static String buildCurl(HttpRequest request, String mimeType, byte[] body)
+    {
+        return getRequestAsString(request)
+                + getHeadersAsString(request.getHeaders())
+                + getContentAsString(mimeType, body);
+    }
+
+    private static String getRequestAsString(HttpRequest request)
+    {
+        String requestString;
+        try
+        {
+            requestString = String.format(CURL_REQUEST_COMMAND, request.getMethod(), request.getUri());
+        }
+        catch (URISyntaxException e)
+        {
+            throw new IllegalArgumentException(e);
+        }
+        return requestString;
+    }
+
+    private static String getHeadersAsString(Header[] headers)
+    {
+        StringBuilder headersString = new StringBuilder();
+        Stream.of(headers).forEach(h -> headersString.append(String.format(HEADER, h.getName(), h.getValue())));
+        return headersString.toString();
+    }
+
+    private static String getContentAsString(String mimeType, byte[] body)
+    {
+        if (body != null)
+        {
+            String bodyAsString = new String(body, StandardCharsets.UTF_8);
+            if (mimeType.contains("multipart"))
+            {
+                String formDataAsString = getFormDataAsString(bodyAsString);
+                return formDataAsString;
+            }
+            return String.format(BODY, bodyAsString);
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private static String getFormDataAsString(String bodyAsString)
+    {
+        String regex = bodyAsString.split("\\R", 2)[0];
+        String[] formDataArray = bodyAsString.split(regex + ".*");
+
+        StringBuilder formDataString = new StringBuilder();
+        Stream.of(formDataArray).forEach(e ->
+        {
+            Matcher m1 = NAME_FILE_NAME_PATTERN.matcher(e);
+            Matcher m2 = NAME_CONTENT_PATTERN.matcher(e);
```

Comment:
if `m1` matches, then `m2` is redundant, please refactor to avoid creation of unused objects